### PR TITLE
refactor: 💡 Adopt HDS copy snippet Desktop UI

### PR DESCRIPTION
**Pull request by calcaide**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/JIRA_TICKET_NUMBER)

## Description

Adopt HDS copy snippet and delete usage of Copyable custom component in Desktop UI.

**BE AWARE**
This PR contains changes that design has to validate and approve first. Here Slack threads about it
- [Thread 1](https://hashicorp.slack.com/archives/C0489FTC7SN/p1708078400823759)
- [Thread 2](https://hashicorp.slack.com/archives/C0489FTC7SN/p1708079301615019)
- [Thread 3](https://hashicorp.slack.com/archives/C0489FTC7SN/p1708079301615019)

## Screenshots:

I selected some random screenshots, too many screens have been change to provide all the screenshots.

### Before

Hidden secret:
hidden_secret_before_open

Sessions list
sessions_before

Targets list
targets_before

Targets list flyout
Screenshot from 2024-02-22 07-18-50

### After

Hidden secret:
hidden_secret_after_open

Sessions list:
Screenshot from 2024-02-22 13-13-06

Targets list:
targets_list_after

Targets list flyout:
flyout_after

## How to Test

I tested manually every template that has been change.

- Navigate to the template.
- Validate the hover state of the Copy::Snippet.
- Click on the the copy button.
- Paste on a different app to validate it works.

## Checklist:

- [ ] I have added before and after screenshots for UI changes
~~- [ ] I have added JSON response output for API changes~~
- [ ] I have added steps to reproduce and test for bug fixes in the description
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
- [ ] My changes generate no new warnings
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~

### CODE REVIEW

File: ui/desktop/app/components/copyable/index.hbs
Comment (ZedLi) [inline ui/desktop/app/components/copyable/index.hbs:1]: Do you know if the usage of `<CopyButton>` can also be replaced with the HDS version or no?

Comment (calcaide) [reply ui/desktop/app/components/copyable/index.hbs:1]: @ZedLi 
As discussed through Slack. We can change it but right now our UI is not using the component part (toolbar) that is using the `<CopyButton>`. I will create a separate task to take care of this.

**Comment by madeleinetle** [comment]
Looks good to me! There are only two comments/questions/changes I have:

1. I would make is that the copy button next to the IDs on the targets page should look the same as those on the sessions page. [The design displaying this can be found here. ](https://www.figma.com/file/lvhT00hEVGZA59PBLlX991/Search-and-Filtering?type=design&node-id=1892%3A114452&mode=design&t=LMNt20DCx9NJ87du-1)
2. Why is the font for the session IDs that are links different from those that aren't links? Can we make them the same font if possible?

File: ui/desktop/app/components/hidden-secret/index.hbs [hunk 1]
```diff
@@ -4,26 +4,25 @@
...
+  <div class='secret-content'>
     {{#if this.isHidden}}
       <pre>{{this.displaySecretValue}}</pre>
     {{else}}
       <span class='secret-mask'>{{this.textMask}}</span>
     {{/if}}
-  </Copyable>
+  </div>
+
+  <Hds::Copy::Button
```
Comment (ZedLi) [inline ui/desktop/app/components/hidden-secret/index.hbs:23]: Should we prevent the height of the buttons from expanding as shown in this big secret?

image

**Review by ZedLi — APPROVED** [review]
Looks good! Thanks for this refactor. Just left one comment regarding the height of the buttons with large secrets

File: ui/desktop/tests/acceptance/projects/targets/index-test.js [hunk 2]
```diff
@@ -106,7 +106,7 @@ module('Acceptance | projects | targets | index', function (hooks) {
       type: 'org',
       scope: globalScope,
     });
-    const org2Scope = { id: instances.scopes.org2.id, type: 'org' };
+    const org2Scope = { id: instances.scopes.org.id, type: 'org' };
```
Comment (cameronperera) [inline ui/desktop/tests/acceptance/projects/targets/index-test.js:109]: non-blocking: I am curious why this was changed? If it does need to be changed, could we remove this variable and use `orgScope` from line 100?

**Review by cameronperera — APPROVED** [review]
Thanks for this large refactor. I left one non-blocking comment.

File: ui/desktop/app/components/hidden-secret/index.hbs (hunk 1 above)
Comment (calcaide) [reply ui/desktop/app/components/hidden-secret/index.hbs:23]: @ZedLi looking into it

Comment (calcaide) [reply ui/desktop/app/components/hidden-secret/index.hbs:23]: Fix it! thanks for the catch!!

File: ui/desktop/tests/acceptance/projects/targets/index-test.js (hunk 2 above)
Comment (cameronperera) [reply ui/desktop/tests/acceptance/projects/targets/index-test.js:109]: I am good with that approach. Thanks for explaining.

**Pull request merged**